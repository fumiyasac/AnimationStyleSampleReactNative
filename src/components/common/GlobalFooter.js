/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import { 
  Footer,
  FooterTab
} from 'native-base';

export default class GlobalFooter extends React.Component {

  // MARK: - Rendering Components

  render() {
    const { tabs } = this.props;
    return (
      <Footer>
        <FooterTab>
          {tabs}
        </FooterTab>
      </Footer>
    );
  };
}
