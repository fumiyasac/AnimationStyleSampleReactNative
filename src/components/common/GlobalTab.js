/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  StyleSheet
} from 'react-native';

import { 
  Button,
  Icon,
  Text
} from 'native-base';

export default class GlobalTab extends React.Component {
  render() {
    return (
      <Button vertical onPress={this.props.onPress}>
        <Icon style={this.props.selected ? styles.tabSelectedColor : styles.tabDefaultColor} name={this.props.icon} />
        <Text style={this.props.selected ? styles.tabSelectedColor : styles.tabDefaultColor}>{this.props.title}</Text>
      </Button>
    );
  }
}

const styles = StyleSheet.create({
  tabDefaultColor: {
    color: '#999999',
  },
  tabSelectedColor: {
    color: '#4169e1'
  }
});