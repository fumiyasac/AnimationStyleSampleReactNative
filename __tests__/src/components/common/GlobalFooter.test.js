import 'react-native';
import React from 'react';
import GlobalFooter from '../../../../src/components/common/GlobalFooter';
import GlobalTab from '../../../../src/components/common/GlobalTab';

import renderer from 'react-test-renderer';

it('GlobalFooterコンポーネントが正しく表示されること', () => {

  // 表示対象のオブジェクト
  let stubObjectArray = [
    { screen: "feed", title: "フィード", icon: "list" },
    { screen: "diary", title: "勉強日記", icon: "clipboard" },
    { screen: "record", title: "達成記録", icon: "medal" },
    { screen: "shopping", title: "お買い物", icon: "book" }
  ];

  // onPressのモック
  const onPressMock = jest.fn();

  // タブのボタンを表示する
  const makeTabs = () => { 
    return stubObjectArray.map( (tabBarItem, index) => {
      return (
        <GlobalTab 
          key={index}
          selected={0 === index}
          title={tabBarItem.title}
          icon={tabBarItem.icon}
          onPress={onPressMock()} />
      );
    });
  };

  // スナップショットの書き出し
  const tree = renderer.create(
    <GlobalFooter
      tabs={makeTabs()} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
  