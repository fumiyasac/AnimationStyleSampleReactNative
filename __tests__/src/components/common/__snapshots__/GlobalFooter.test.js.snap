// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFooterコンポーネントが正しく表示されること 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F8F8",
      "borderColor": "#cbcbcb",
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexDirection": "row",
      "height": 55,
      "justifyContent": "center",
      "left": 0,
      "paddingBottom": 0,
      "right": 0,
    }
  }
  virtual={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": undefined,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#6b6b6b",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
              Object {
                "color": "#4169e1",
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "color": "#4169e1",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        フィード
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#6b6b6b",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
              Object {
                "color": "#999999",
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "color": "#999999",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        勉強日記
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#6b6b6b",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
              Object {
                "color": "#999999",
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "color": "#999999",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        達成記録
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flex": 1,
          "flexDirection": "column",
          "height": null,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#6b6b6b",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
              Object {
                "color": "#999999",
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        virtual={undefined}
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#6b6b6b",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            Object {
              "color": "#999999",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      >
        お買い物
      </Text>
    </View>
  </View>
</View>
`;
